packages/nx/src/native/cache/http_remote_cache.rs

```diff
@@ -0,0 +1,206 @@
+        terminal_output_header.set_cksum(); // Ensure the checksum is set correctly
+        archive
+            .append_data(
+                &mut terminal_output_header,
+                "terminalOutput",
+                terminal_output_bytes,
+            )
+            .unwrap();
+        trace!("Added terminal output to tarball");
+
+        trace!("Adding code to tarball");
+        let mut code_header = tar::Header::new_old();
+        code_header.set_size(4);
+        code_header.set_cksum(); // Ensure the checksum is set correctly
+        archive
+            .append_data(&mut code_header, "code", &code.to_be_bytes()[..])
+            .unwrap();
+        trace!("Added code to tarball");
+
+        trace!("Finishing tarball");
+        archive
+            .finish() // Finish the archive to get the inner bytes
+            .map_err(|e| anyhow::anyhow!(format!("Failed to finish tarball: {}", e)))?;
+        trace!("Finished tarball");
+
+        trace!("Reading tarball into memory");
+        let archive_bytes = archive.into_inner().unwrap();
+        let buffer = archive_bytes.finish()?;
+        // let mut buffer = Vec::new();
+        // // read the whole file
+        // archive_bytes.read_to_end(&mut buffer)?;
+        trace!("read tarball into memory");
+
+        let url: String = format!("{}/v1/cache/{}", self.url, hash);
+        let response = self
+            .client
+            .put(&url)
+            .header(header::CONTENT_TYPE, "application/octet-stream")
+            .body(buffer) // Convert the bytes to a Vec<u8> for the request body
+            .send()
+            .await
+            .map_err(|e| anyhow::anyhow!("Failed to send request: {}", e))?;
+
+        Ok(response.status().is_success())
+    }
+
+    async fn download_and_extract_from_result(
+        response: reqwest::Response,
+        cache_directory: String,
+    ) -> anyhow::Result<CachedResult> {
+        let content = response.bytes().await.unwrap();
+        trace!("Downloaded {} bytes from remote cache", content.len());
+        let tar = flate2::read::GzDecoder::new(content.as_ref());
+        let mut archive = Archive::new(tar);
+        let entries = archive
+            .entries() // Get the entries in the archive
+            .map_err(|_| anyhow::anyhow!("Failed to read entries from tarball"))?;
+
+        // Extract the archive to the specified cache directory
+        for entry in entries {
+            let mut entry =
+                entry.map_err(|_| anyhow::anyhow!("Failed to read entry from tarball"))?;
+            trace!(
+                "Extracting entry: {}",
+                entry
+                    .path()
+                    .map(|p| p.to_string_lossy().into_owned())
+                    .unwrap_or_else(|_| "<unknown>".to_string())
+            );
+            // Ensure the directory exists before extracting
+            if let Err(e) = entry.unpack_in(&cache_directory) {
```

Comment:
Did we want to ensure the directory? Is something missing here?

Reply:
`unpack_in` does ensure dir on its own, but we actually needed to swap to `unpack` after this review so I added some ensure dir.